tests/components/rainmachine/test_diagnostics.py

```diff
@@ -0,0 +1,592 @@
+"""Test RainMachine diagnostics."""
+from homeassistant.components.diagnostics import REDACTED
+
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_entry_diagnostics(hass, config_entry, hass_client, setup_rainmachine):
+    """Test config entry diagnostics."""
+    assert await get_diagnostics_for_config_entry(hass, hass_client, config_entry) == {
```

Comment:
Would it make sense to refer to the json fixtures in the asserted response instead of repeating it all here?

Reply:
Great question! I'm always torn on that – part of me thinks that we should be testing the exact output we expect, word for word, but I'm open!